ouroboros-consensus-byron/src/Ouroboros/Consensus/Byron/Ledger/Forge.hs

```diff
@@ -156,22 +166,34 @@ forgeRegularBlock cfg bno sno st txs isLeader =
     updatePayload = CC.Update.payload (bpUpProposal blockPayloads)
                                       (bpUpVotes blockPayloads)
 
-    extendBlockPayloads :: GenTx ByronBlock
+    extendBlockPayloads :: Word32
+                        -> GenTx ByronBlock
                         -> BlockPayloads
                         -> BlockPayloads
-    extendBlockPayloads genTx bp@BlockPayloads{bpTxs, bpDlgCerts, bpUpVotes} =
-      -- TODO: We should try to use 'recoverProof' (and other variants of
-      -- 'recoverBytes') here as opposed to throwing away the serializations
-      -- (the 'ByteString' annotations) with 'void' as we're currently doing.
-      case genTx of
-        ByronTx             _ tx   -> bp { bpTxs        = void tx : bpTxs }
-        ByronDlg            _ cert -> bp { bpDlgCerts   = void cert : bpDlgCerts }
-        -- TODO: We should throw an error if we encounter multiple
-        -- 'ByronUpdateProposal's (i.e. if 'bpUpProposal' 'isJust').
-        -- This is because we should only be provided with a maximum of one
-        -- 'ByronUpdateProposal' to include in a block payload.
-        ByronUpdateProposal _ prop -> bp { bpUpProposal = Just (void prop) }
-        ByronUpdateVote     _ vote -> bp { bpUpVotes    = void vote : bpUpVotes }
+    extendBlockPayloads maxBlockSize genTx bp@BlockPayloads{bpTxs, bpDlgCerts, bpUpVotes, bpSize}
+      -- do not include transaction if it will exceed maximum block size
+      | bpSize + txInBlockSize genTx > maxBlockSize = bp
+      | otherwise                                   =
+        -- TODO: We should try to use 'recoverProof' (and other variants of
+        -- 'recoverBytes') here as opposed to throwing away the serializations
+        -- (the 'ByteString' annotations) with 'void' as we're currently doing.
+        case genTx of
```

Comment:
It would be more terse to introduce a `bp' = bp { bpSize = bpSize + txInBlockSize genTx }` and then update `bp'` in each branch. That makes it clear that regardless of the `genTx` constructor, we update the `bpSize` field the same way.

What do you think?

Reply:
(You can also do `(\bp' -> bp' { bpSize = bpSize + txInBlockSize genTx }) $ case genTx of ...` or the other way around, if you like. That approach makes it impossible to accidentally use `bp` instead of `bp'` in one of the case alternatives.